File: U2/T.I.2.1 PAO202551_EUREKABANK_BDDNUBE_SOAP_JAVA_GR01/04. SERVIDOR/EurekaBank_Soap_Java_GR01/src/java/ec/edu/monster/constants/MensajesConstants.java
package ec.edu.monster.constants;

/**
 * Constantes para mensajes del sistema
 */
public class MensajesConstants {

    // Mensajes de éxito
    public static final String LOGIN_EXITOSO = "Autenticación exitosa";
    public static final String REGISTRO_EXITOSO = "Empleado registrado correctamente";
    public static final String CLAVE_ACTUALIZADA = "Clave actualizada correctamente";
    public static final String DEPOSITO_EXITOSO = "Depósito realizado correctamente";
    public static final String RETIRO_EXITOSO = "Retiro realizado correctamente";
    public static final String TRANSFERENCIA_EXITOSA = "Transferencia realizada correctamente";

    // Mensajes de error - Autenticación
    public static final String CREDENCIALES_INVALIDAS = "Usuario o clave incorrectos";
    public static final String USUARIO_INACTIVO = "Usuario inactivo";
    public static final String USUARIO_EXISTE = "El usuario ya existe";
    public static final String CLAVE_ACTUAL_INCORRECTA = "La clave actual es incorrecta";

    // Mensajes de error - Cuenta
    public static final String CUENTA_NO_EXISTE = "La cuenta no existe";
    public static final String CUENTA_INACTIVA = "La cuenta está inactiva";
    public static final String CLAVE_INCORRECTA = "Clave de cuenta incorrecta";
    public static final String SALDO_INSUFICIENTE = "Saldo insuficiente";
    public static final String MONEDA_DIFERENTE = "Las cuentas deben ser de la misma moneda";
    public static final String CUENTA_DESTINO_INVALIDA = "Cuenta destino inválida";
    public static final String MISMA_CUENTA = "No se puede transferir a la misma cuenta";

    // Mensajes de error - Validación
    public static final String IMPORTE_INVALIDO = "El importe debe ser mayor a cero";
    public static final String DATOS_INCOMPLETOS = "Datos incompletos";
    public static final String EMPLEADO_NO_EXISTE = "El empleado no existe";

    // Mensajes de error - Sistema
    public static final String ERROR_TRANSACCION = "Error al procesar la transacción";
    public static final String ERROR_BASE_DATOS = "Error al acceder a la base de datos";
    public static final String ERROR_INESPERADO = "Error inesperado del sistema";

    // Constructor privado para evitar instanciación
    private MensajesConstants() {
        throw new AssertionError("No se puede instanciar esta clase");
    }
}

File: U2/EurekaBank_REST_GoogleMaps/EurekaBank_RestFull_DotNet_GR01/EurekaBank_RestFull_DotNet_GR01/constants/MensajesConstants.cs
namespace EurekaBank_RestFull_DotNet_GR01.Constants
{
    /// <summary>
    /// Mensajes estándar para respuestas del sistema
    /// </summary>
    public static class MensajesConstants
    {
        // Mensajes de éxito
        public const string DEPOSITO_EXITOSO = "Depósito realizado exitosamente";
        public const string RETIRO_EXITOSO = "Retiro realizado exitosamente";
        public const string TRANSFERENCIA_EXITOSA = "Transferencia realizada exitosamente";
        public const string LOGIN_EXITOSO = "Autenticación exitosa";
        public const string REGISTRO_EXITOSO = "Empleado registrado exitosamente";
        public const string SUCURSAL_CREADA = "Sucursal creada correctamente";
        public const string SUCURSAL_ACTUALIZADA = "Sucursal actualizada correctamente";
        public const string SUCURSAL_ELIMINADA = "Sucursal eliminada correctamente";
        public const string CLAVE_CAMBIADA = "Contraseña cambiada exitosamente";
        
        // Errores de validación
        public const string ERROR_CAMPOS_REQUERIDOS = "Uno o más campos requeridos están vacíos";
        public const string ERROR_MODELO_INVALIDO = "Los datos proporcionados no son válidos";
        public const string ERROR_FORMATO_INVALIDO = "El formato de los datos no es válido";
        public const string ERROR_DATOS_NULOS = "Los datos proporcionados no pueden ser nulos";
        public const string ERROR_CAMPOS_MINIMOS = "Debe proporcionar al menos un campo para actualizar";
        public const string ERROR_RANGO_INVALIDO = "El valor está fuera del rango permitido";
        public const string ERROR_LONGITUD_INVALIDA = "La longitud del campo no es válida";
        
        // Errores de cuenta
        public const string ERROR_CUENTA_NO_EXISTE = "La cuenta no existe";
        public const string ERROR_CUENTA_INACTIVA = "La cuenta no está activa";
        public const string ERROR_CLAVE_INCORRECTA = "Clave incorrecta";
        public const string ERROR_SALDO_INSUFICIENTE = "Saldo insuficiente para realizar la operación";
        
        // Errores de transacciones
        public const string ERROR_IMPORTE_INVALIDO = "El importe debe ser mayor a cero";
        public const string ERROR_IMPORTE_DECIMALES = "El importe no puede tener más de 2 decimales";
        public const string ERROR_MONEDA_DIFERENTE = "Las cuentas deben ser de la misma moneda";
        
        // Errores de autenticación
        public const string ERROR_CREDENCIALES_INVALIDAS = "Usuario o contraseña incorrectos";
        public const string ERROR_USUARIO_EXISTENTE = "El usuario ya existe";
        public const string ERROR_USUARIO_VACIO = "El usuario no puede estar vacío";
        public const string ERROR_CLAVE_CORTA = "La contraseña debe tener al menos 6 caracteres";
        public const string ERROR_USUARIO_NO_ENCONTRADO = "Usuario no encontrado";
        
        // Errores de sucursales
        public const string ERROR_SUCURSAL_NO_ENCONTRADA = "Sucursal no encontrada";
        public const string ERROR_SUCURSAL_TIENE_CUENTAS = "No se puede eliminar la sucursal porque tiene cuentas asociadas";
        public const string ERROR_CODIGO_SUCURSAL_INVALIDO = "El código de sucursal debe ser mayor a cero";
        public const string ERROR_COORDENADAS_INCONSISTENTES = "Las coordenadas deben ser ambas nulas o ambas tener valor válido";
        
        // Errores de recursos
        public const string ERROR_RECURSO_NO_ENCONTRADO = "El recurso solicitado no fue encontrado";
        public const string ERROR_RECURSO_YA_EXISTE = "El recurso ya existe";
        public const string ERROR_RECURSO_EN_USO = "El recurso está siendo utilizado y no puede ser eliminado";
        
        // Errores generales del sistema
        public const string ERROR_DATOS_INCOMPLETOS = "Faltan datos obligatorios";
        public const string ERROR_OPERACION_FALLIDA = "La operación no pudo completarse";
        public const string ERROR_BASE_DATOS = "Error al conectar con la base de datos";
        public const string ERROR_SERVIDOR_INTERNO = "Error interno del servidor";
        public const string ERROR_TIMEOUT = "La operación excedió el tiempo límite";
    }
}
